src/protocol/produce.rs

```diff
@@ -257,31 +257,35 @@ pub struct PartitionProduceResponse {
 }
 
 impl ProduceResponse {
-    pub fn get_response(&self) -> Vec<TopicPartitionOffset> {
+    pub fn get_response(self) -> Vec<ProduceConfirm> {
         self.topic_partitions
-            .iter()
-            .flat_map(|ref tp| tp.get_response(tp.topic.clone()))
+            .into_iter()
+            .map(|tp| tp.get_response())
             .collect()
     }
 }
 
 impl TopicPartitionProduceResponse {
-    pub fn get_response(&self, topic: String) -> Vec<TopicPartitionOffset> {
-        self.partitions
+    pub fn get_response(self) -> ProduceConfirm {
+        let confirms = self.partitions
             .iter()
-            .map(|ref p| p.get_response(topic.clone()))
```

Comment:
You think it's ok to make this method consuming? I did it to get rid of the need to clone, but I wasn't sure what you'd think.